New `test_autoscaler.py::test_autoscaler_requeues_failed_tasks` test flaky?

While testing ≈unrelated #1690:

Attempt 1 [failed]:
```
FAILED tests/taggers/test_autoscaler.py::test_autoscaler_requeues_failed_tasks - AssertionError: Task was not dispatched to the initial actor
assert False
```

Attempt 2 [passed].

Looks like it was added today in #1769, cc @BabyChouSr

[failed]: https://github.com/marin-community/marin/actions/runs/18581248506/job/52976576248?pr=1690#step:8:1214
[passed]: https://github.com/marin-community/marin/actions/runs/18581248506/job/52976908745?pr=1690#step:8:900

--- a/src/marin/processing/classification/autoscaler.py
+++ b/src/marin/processing/classification/autoscaler.py
@@ -193,8 +193,8 @@ class AutoscalingActorPool:
                     break
 
             if not dispatched_any:
-                logger.info("No tasks dispatched - going to sleep for 10 seconds.")
-                time.sleep(10)
+                logger.info("No tasks dispatched - going to sleep for 1 second.")
+                time.sleep(1)
 
     def _result_collector_loop(self):
         """Start a background thread that collects completed futures."""

--- a/tests/taggers/test_autoscaler.py
+++ b/tests/taggers/test_autoscaler.py
@@ -50,7 +50,7 @@ def test_autoscaler_requeues_failed_tasks(ray_tpu_cluster):
         while time.time() < deadline:
             with pool.actor_task_metadata_lock:
                 pending = pool.actor_futures.get(initial_actor, [])
-            if pending:
+            if pending or result_queue.qsize() > 0:  # pending task or finished task
                 task_assigned = True
                 break
             time.sleep(0.05)
@@ -69,9 +69,11 @@ def test_autoscaler_requeues_failed_tasks(ray_tpu_cluster):
             processed_task = result_queue.get(timeout=30)
             break
 
+        print("Finished processing first task")
+
         assert processed_task["id"] == original_task["id"]
 
         with pytest.raises(ray.exceptions.RayTaskError):
-            result_queue.get(timeout=30)  # next one should not succeed since only one task
+            result_queue.get(timeout=5)  # next one should not succeed since only one task
     finally:
         pool.shutdown()

